xds/internal/client/v2client_rds.go

```diff
@@ -94,42 +95,154 @@ func generateRDSUpdateFromRouteConfiguration(rc *xdspb.RouteConfiguration, host
 		// should be at least one default route.
 		return rdsUpdate{}, fmt.Errorf("matched virtual host has no routes")
 	}
-	dr := vh.Routes[len(vh.Routes)-1]
-	match := dr.GetMatch()
-	if match == nil {
-		return rdsUpdate{}, fmt.Errorf("matched virtual host's default route doesn't have a match")
-	}
-	if prefix := match.GetPrefix(); prefix != "" && prefix != "/" {
-		// The matched virtual host is invalid. Match is not "" or "/".
-		return rdsUpdate{}, fmt.Errorf("matched virtual host's default route is %v, want Prefix empty string or /", match)
-	}
-	if caseSensitive := match.GetCaseSensitive(); caseSensitive != nil && !caseSensitive.Value {
-		// The case sensitive is set to false. Not set or set to true are both
-		// valid.
-		return rdsUpdate{}, fmt.Errorf("matched virtual host's default route set case-sensitive to false")
+
+	// Keep the old code path for routing disabled.
+	if !routingEnabled {
```

Comment:
Style nit:
You could either make a helper function for the code in here, or if you don't prefer that (since it is not called from multiple places), the code flow should be the other way around:

if routingEnabled {
  // New code
  return
}
// Old code

This will not only reduce the amount of diff, but it also makes sure that code is indented less.